pkg/daemon/ceph/osd/nsenter.go

```diff
@@ -72,10 +73,20 @@ func (ne *NSEnter) callNsEnter(binPath string) error {
 func (ne *NSEnter) checkIfBinaryExistsOnHost() error {
 	for _, path := range binPathsToCheck {
 		binPath := filepath.Join(path, ne.binary)
+		// Check with nsenter first
 		err := ne.callNsEnter(binPath)
 		if err != nil {
 			logger.Debugf("failed to call nsenter. %v", err)
-			continue
+			// If nsenter failed, let's try with the rootfs directly
+			rootFSBinPath := filepath.Join(rootFSPath, binPath)
+			output, err := ne.context.Executor.ExecuteCommandWithCombinedOutput(rootFSBinPath, ne.binaryArgs...)
+			if err != nil {
+				logger.Debugf("failed to execute command on the host rootfs. %s. %v", output, err)
+				continue
+			} else {
+				logger.Infof("binary %q found on the host rootfs, proceeding with osd preparation", rootFSBinPath)
```

Comment:
You don't need this else block on lines 86-88. It would just fall through to return successfully on line 92, right?

Reply:
Yes but I get a proper success message with this `else`.

Reply:
Yes it works too, it's just duplicated code.

Reply:
Although I see no that the path in the message is different

Reply:
Line87 shows "/rootfs/sbin/lvm" but Line 91 shows "/sbin/lvm". How about the following more simple code?.

```
                        if err != nil {
				logger.Debugf("failed to execute command on the host rootfs. %s. %v", output, err)
				continue
			}
			binPath = rootFSBinPath
			}
             }
             logger.Infof("binary %q found on the host, proceeding with osd preparation", binPath)
```

Reply:
One has `logger.Infof("binary %q found on the host rootfs, proceeding with osd preparation", rootFSBinPath)` there other has `logger.Infof("binary %q found on the host, proceeding with osd preparation", binPath)`.

Are we talking about the same thing?

Reply:
How about the following more simple code? It can avoid the duplicated message.

```
                        if err != nil {
				logger.Debugf("failed to execute command on the host rootfs. %s. %v", output, err)
				continue
			}
			binPath = rootFSBinPath
			}
             }
             logger.Infof("binary %q found on the host, proceeding with osd preparation", binPath)
```